examples/tv-casting-app/android/App/app/src/main/jni/cpp/ConversionUtils.cpp

```diff
@@ -155,8 +157,33 @@ CHIP_ERROR convertJVideoPlayerToTargetVideoPlayerInfo(jobject videoPlayer, Targe
     jstring jHostName         = static_cast<jstring>(env->GetObjectField(videoPlayer, getHostNameField));
     const char * hostName     = env->GetStringUTFChars(jHostName, 0);
 
+    jfieldID jPort = env->GetFieldID(jVideoPlayerClass, "port", "I");
+    uint16_t port  = static_cast<uint16_t>(env->GetIntField(videoPlayer, jPort));
+
+    jfieldID getInstanceNameField = env->GetFieldID(jVideoPlayerClass, "instanceName", "Ljava/lang/String;");
+    jstring jInstanceName         = static_cast<jstring>(env->GetObjectField(videoPlayer, getInstanceNameField));
+    const char * instanceName     = env->GetStringUTFChars(jInstanceName, 0);
+
+    jfieldID jLastDiscoveredMs = env->GetFieldID(jVideoPlayerClass, "lastDiscoveredMs", "J");
+    long lastDiscoveredMs      = static_cast<long>(env->GetLongField(videoPlayer, jLastDiscoveredMs));
+
+    jfieldID getMACAddressField = env->GetFieldID(jVideoPlayerClass, "MACAddress", "Ljava/lang/String;");
+    jstring jMACAddress         = static_cast<jstring>(env->GetObjectField(videoPlayer, getMACAddressField));
+    const char * MACAddress     = env->GetStringUTFChars(jMACAddress, 0);
+
+    jfieldID jIsAsleep = env->GetFieldID(jVideoPlayerClass, "isAsleep", "Z");
+    bool isAsleep      = static_cast<bool>(env->GetLongField(videoPlayer, jIsAsleep));
+
     outTargetVideoPlayerInfo.Initialize(nodeId, fabricIndex, nullptr, nullptr, vendorId, productId, deviceType, deviceName,
-                                        hostName);
+                                        hostName, 0, nullptr, port, instanceName, chip::System::Clock::Timestamp(lastDiscoveredMs));
+
+    if (MACAddress != nullptr)
```

Comment:
how come this is not guaranteed?

Reply:
A VideoPlayer may indeed choose to not reveal its MACAddress when we try to read it from its WakeOnLAN cluster (which itself is optionally supported). Therefore, we need this condition here.